### ISSUE TITLE
Cross-compile on Linux for Windows ARM with clang-cl

### ISSUE THREAD

**Issue by t5555d** [openssl/openssl#12363]
I am trying to build openssl for "Windows 10 for ARM" on Linux build agent with clang-cl.
There is predefined `VC-WIN64-ARM` configuration, which I successfully utilized to build with MSVC on Windows.
With some additional options I was also able to build with clang-cl / llvm-lib / lld-link on Windows.

Approximately this was the recipe:
```
set CFLAGS=--target=aarch64-windows-msvc -imsvc ...
set ARFLAGS=/MACHINE:ARM64
set LDFLAGS=/MACHINE:ARM64 -libpath:...
perl Configure --release VC-WIN64-ARM -O2 CC=clang-cl AR=llvm-lib LD=lld-link no-shared
nmake
```

But when I use such recipe to cross-compile on Linux, it fails with such message:
```
This perl implementation doesn't produce Windows like paths (with backward
slash directory separators).  Please use an implementation that matches your
building platform.

This Perl version: 5.26.1 for x86_64-linux-gnu-thread-multi
```

Obviously `VC-WIN64-ARM` works only on Windows.
I haven't found any other configuration for Windows ARM target.
Is it possible to cross-compile openssl for Windows ARM on Linux?

**Comment by mattcaswell** [comment]
Hmm. Interesting it seems you are using clang-cl and using one of the Visual Studio config targets. Normally when cross-compiling for Windows on Linux we use the mingw compilers and the mingw targets. I have no idea whether this is even possible using clang-cl on Linux.

Perhaps @levitte might know.

**Comment by bernd-edlinger** [comment]
you can try to set `CONFIGURE_INSIST=yes` before invoking Configure.

**Comment by mattcaswell** [comment]
Although I suspect you will end up with an "nmake" style Makefile with that Configure target.

**Comment by levitte** [comment]
> Although I suspect you will end up with an "nmake" style Makefile with that Configure target.

Yup.  "VC" implies Visual C with all the corresponding tools, and has never been adapted for cross compilation in this sense.

clang blurs the line, it seems...

**Comment by nhorman** [comment]
Marking as inactive to be closed at the end of 3.4 dev, barring further input

**Comment by t8m** [comment]
Closing as the issue was labeled as inactive for extended period of time and there was no additional input.
Please add a comment if you have any further information to support reopening the issue.
